git-webui/src/share/git-webui/webui/js/git-webui.js

```diff
@@ -3012,34 +3013,88 @@ webui.NewChangedFilesView = function(workspaceView) {
     }
 
     self.amend = function(message, details) {
-
         if (self.commitMsgEmpty()) {
-            webui.git_command(["add"].concat(selectedItems));
-            webui.git_command(['commit', '--amend', '--no-edit', '--'].concat(selectedItems), function(output) {
-                webui.showSuccess(output);
-                workspaceView.update();
+            $.ajax({
+                url: "git-command-unstage",
+                type: "POST",
+                contentType: 'application/json',
+                data: JSON.stringify({
+                    command: "commit", 
+                    message: message, 
+                    details: details, 
+                    flags: "--amend,--no-edit",
+                    files: JSON.stringify({
```

Comment:
The nested JSON.stringify is gross here. Should just pass files (top level) as an array - any reason not to?